spaceship.zsh

```diff
@@ -738,6 +746,25 @@ spaceship_julia() {
     "$SPACESHIP_JULIA_SUFFIX"
 }
 
+# JAVA
+# Show current Java version
+spaceship_java() {
+  [[ $SPACESHIP_JAVA_SHOW == false ]] && return
+
+  _exists java || return
+
+  # Show Java version only for Java-related folders
+  [[ -f pom.xml ]] || return
+
+  local java_version=$(java -version 2>&1 | head -n 1 | awk -F '"' '{print $2}')
```

Comment:
Is there any Java version manager?

Reply:
I use sdkman and frequently switch java versions using it. 
It would be great if you could support it

Reply:
This line won't work if the variable `JAVA_TOOL_OPTIONS` is set as java prints that it picked the variable :

```
# JAVA_TOOL_OPTIONS="-Dfile.encoding=UTF-8" java -version
Picked up JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF-8
java version "1.8.0_144"
Java(TM) SE Runtime Environment (build 1.8.0_144-b01)
Java HotSpot(TM) 64-Bit Server VM (build 25.144-b01, mixed mode)
```

Reply:
Also why asking _jenv_ or _sdkman_ or whatever, they maybe useful, but what is relevant is the java version that is available in the shell, right ?

With jenv you can set your version globally, in the shell, or locally (in folder and sub-folders), jenv will pick the configured version just by executing the java in the path. Spaceship could do the same.

Reply:
Should be OK to change to:
```sh
local java_version=$(java -version 2>&1 | grep '^java version ' | awk -F '"' '{print $2}')
```

Will do when I re-work this for `3.x`.

And yeah, I agree. Those tools should change which binary is available on the path, right? So it feels like asking a potential version manager is redundant?